modules/core/src/main/java/org/apache/ignite/metric/IgniteMetricRegistry.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.metric;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+import java.util.function.IntSupplier;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+import org.apache.ignite.lang.IgniteExperimental;
+import org.apache.ignite.spi.metric.ReadOnlyMetricRegistry;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Metric registry. Allows to get, add or remove metrics.
+ *
+ * @see IgniteMetrics
+ * @see ReadOnlyMetricRegistry
+ */
+@IgniteExperimental
+public interface IgniteMetricRegistry extends ReadOnlyMetricRegistry {
```

Comment:
I doubt about naming here - `IgniteMetricRegistry` is something that belongs to Ignite.
But, here it used for registry that will be managed and fullfillled by the user.

maybe we can use just `MetricRegistry` here.
And rename internal implementation to `MetricRegistryImpl`?

Reply:
@Vladsz83 Let's decide on this suggestion before proceed with the merge.

Reply:
We've discussed remaning internal MetricRegistry -> MetricRegistryImpl here : https://github.com/apache/ignite/pull/11216/files . Also, we tried temporary keep two internal and external interfaces MetricRegistry to avoid related tons of code cnahges :  https://github.com/apache/ignite/pull/11221/files . The decision was that twe need more exposed, available for internal usage metric interfaces, internal code remenings and code cleanup before using 'MetricRegistry'. Since we declare the interface @IgniteExperimental, we could rename it later.

Reply:
Also, we plan to expose internal read-only metrics as noted before using the same API, ` findRegistry(String name)`. They belong to Ignite.

Reply:
> Also, we plan to expose internal read-only metrics as noted before using the same API

Then method must return `ReadOnlyMetricRegistry`.

> to avoid related tons of code cnahges

I don't think "tons of changes" is the right reason to use bad naming.
Let's make API more clear by using clear naming.

Reply:
>> Then method must return ReadOnlyMetricRegistry.

Yes, but not IgniteReadOnlyMetricRegistry. Ok, I'll try to rename and make use MetricRegistry (not MetricRegistryImpl) as much as possible

Reply:
Ok, I prepared another PR with the renaming internal `class MetricRegistry` -> `class MetricRegistryImpl` and new `interface IgniteMetricRegistry` -> `interface MetricRegistry` : https://github.com/apache/ignite/pull/11293/files . 
Used `MetricRegistry` instead of `MetricRegistryImpl` where is possible. We planned to add methods like `longMetric` to the new registry interface in a next PR. Then, we could refactor to more internal usage of `interface MetricRegistry`.

I recalled one problem that we discussed: `CdcConsumer`. It is `@IgniteExperimental` but also a public API. And this feature look crucial. It uses internal `org.apache.ignite.internal.processors.metric.MetricRegistry`. Renaming to `MetricRegistryImpl` will break at least the extensions. And any external code which uses `CdcConsumer` and probably related `org.apache.ignite.internal.processors.metric.MetricRegistry` (`CdcConsumer#start(MetricRegistry mreg)`). IMHO, keeping `@IgniteExperimental IgniteMetricRegistry` for a while and possible later renaming to `@IgniteExperimental MetricRegistry` is a less evil. WDYT?